stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt

```diff
@@ -18,83 +20,121 @@ import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.client.utils.internal.toggle.ToggleService
 import io.getstream.chat.android.core.ExperimentalStreamChatApi
-import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelState
+import io.getstream.chat.android.offline.experimental.extensions.asReferenced
+import io.getstream.chat.android.offline.experimental.extensions.globalState
 import io.getstream.chat.android.offline.extensions.setMessageForReply
+import kotlinx.coroutines.flow.combine
 import java.io.File
 
 /**
  * ViewModel class for MessageInputView. Responsible for sending and updating chat messages.
  * Can be bound to the view using the MessageInputViewModel.bindView function.
+ *
  * @param cid The full channel id, i.e. "messaging:123".
- * @param chatDomain Entry point for all livedata & offline operations.
  * @param chatClient Entry point for most of the chat SDK
  */
 public class MessageInputViewModel @JvmOverloads constructor(
     private val cid: String,
-    private val chatDomain: ChatDomain = ChatDomain.instance(),
     private val chatClient: ChatClient = ChatClient.instance(),
 ) : ViewModel() {
+
+    /**
+     * Holds information about the current channel and is actively updated.
+     */
+    public val channelState: ChannelState =
+        chatClient.asReferenced().watchChannel(cid, MessageListViewModel.MESSAGES_LIMIT).asState(viewModelScope)
+
+    /**
+     * A list of [Channel] members.
+     */
+    public val members: LiveData<List<Member>> = channelState.members.asLiveData()
+
+    /**
+     * List of available commands.
+     */
+    public val commands: LiveData<List<Command>> =
+        Transformations.map(channelState.channelConfig.asLiveData()) { config ->
+            config.commands
+        }
+
+    /**
+     * The cooldown interval for the given channel.
+     */
+    public val cooldownInterval: LiveData<Int> =
+        Transformations.map(channelState.channelData.asLiveData()) { channelData ->
+            channelData.cooldown
+        }
+
+    /**
+     * The maximum length of a message that can be typed in the message input.
+     */
+    public val maxMessageLength: LiveData<Int> =
+        Transformations.map(channelState.channelConfig.asLiveData()) { config ->
```

Comment:
What about use the .map extension function? 
```
channelState.channelConfig.asLiveData().map { config -> config.commands }
```

It is just suggar syntax, of course, but I think it is more aligned with the other Kotlin APIs (like Iterables)

Reply:
This is good advice.

Honestly, I know `LiveData` is usually operator poor so I forget that it has at least some operator capabilities.